src/uct/tcp/tcp.h

```diff
@@ -12,10 +12,62 @@
 
 #define UCT_TCP_NAME "tcp"
 
-
 /* How many events to wait for in epoll_wait */
 #define UCT_TCP_MAX_EVENTS        16
 
+#define UCT_TCP_EP_CONN_STATES(_FUNC) \
+    _FUNC(CLOSED, "connection closed", \
```

Comment:
imo, looks unnecessary complicated. 
would it be much more code if you just declare arrays in place?

Reply:
will think about how to simplify this code. this was implemented to be able to expand this easily because the new states will be added in the new PRs.

Reply:
Ok, then. Just add a TODO: revise (or simplify) for now

Reply:
seems overly complex to me too (@dmitrygladkov seems you have extra love for macros)
how about defining all state properties in single place with single struct, e.g
```
struct uct_tcp_state {
   const char *name;
   ... progress_rx;
   ... progress_tx;
   ...
}
```

Reply:
yes, this is a good idea!

Defined a single struct:
```
struct uct_tcp_cm_state {
      const char            *name;
      const char            *description;
      uct_tcp_ep_progress_t progress[UCT_TCP_EP_CTX_TYPE_LAST];
};
```

> (@dmitrygladkov seems you have extra love for macros)

:)